test/helpers/erc4337.js

```diff
@@ -11,48 +11,51 @@ const parseInitCode = initCode => ({
 /// Global ERC-4337 environment helper.
 class ERC4337Helper {
   constructor() {
-    this.envAsPromise = Promise.all([ethers.provider.getNetwork(), ethers.deployContract('Create2Mock')]).then(
-      ([{ chainId }, factory]) => ({
-        chainId,
-        factory,
-      }),
-    );
+    this.factoryAsPromise = ethers.deployContract('Create2Mock');
   }
 
   async wait() {
-    return (this.env = await this.envAsPromise);
+    this.factory = await this.factoryAsPromise;
+    return this;
   }
 
+  // TODO: Pass the entrypoint in the params (senderCreator has to be discovered accordingly)
   async newAccount(name, extraArgs = [], params = {}) {
```

Comment:
I'd just do:

```suggestion
  async newAccount(name, extraArgs = [], params = {
    entrypoint: entrypoint.v08,
    senderCreator: senderCreator.v08,
  }) {
```

Reply:
If the two are inconsitent we accept that behavior is unexpected ?

Honeslt we don't need that now. We don't have any test set-up with anything other than v08

Reply:
That syntax does not work. If a param object is provided that doesn't include entrypoint/senderCreator, then the default values are not added.

Implemented a bit differently

Reply:
Perfect, I like the alternative better:

```js
const env = {
  entrypoint: params.entrypoint ?? entrypoint.v08,
  senderCreator: params.senderCreator ?? senderCreator.v08,
};
```